Cairo Desktop/CairoDesktop.AppGrabber/TaskCategoryProvider.cs

```diff
@@ -73,23 +79,69 @@ private void CategoryList_CollectionChanged(object sender, System.Collections.Sp
             }
         }
 
-        private ApplicationInfo GetTaskApplicationInfo(ApplicationWindow window)
+        private string GetCategoryDisplayName(ApplicationWindow window)
         {
-            ApplicationInfo appInfo = null;
-            foreach (ApplicationInfo ai in _appGrabber.CategoryList.FlatList)
-            {
-                if (!string.IsNullOrEmpty(ai.Target) && ((!window.IsUWP && ai.Target.ToLower() == window.WinFileName.ToLower()) || (window.IsUWP && ai.Target == window.AppUserModelID)) && ai.Category != null)
-                {
-                    appInfo = ai;
-                    break;
-                }
-                else if (window.Title.ToLower().Contains(ai.Name.ToLower()) && ai.Category != null)
+            if (TryGetCategoryDisplayNameByAppUserModelId(window, out var category)) return category;
+            if (TryGetCategoryDisplayNameByWinFileName(window, out category)) return category;
+            if (TryGetCategoryDisplayNameByProcId(window, out category)) return category;
+            if (TryGetCategoryDisplayNameByFileEquality(window, out category)) return category;
+            if (TryGetCategoryDisplayNameByTitle(window, out category)) return category;
+            return null;
+        }
+
+        private bool TryGetCategoryDisplayNameByAppUserModelId(ApplicationWindow window, out string category)
+        {
+            category = null;
+            if (!window.IsUWP) return false;
```

Comment:
Might be worth leaving a comment here that only UWP apps have an AppUserModelId (or rename the method?)